sklearn/tree/tests/test_tree.py

```diff
@@ -831,7 +841,63 @@ def test_min_impurity_split():
                                       est.min_impurity_split))
 
 
-def test_pickle():
+def test_min_impurity_decrease():
+    # test if min_impurity_decrease ensure that a split is made only if
+    # if the impurity decrease is atleast that value
+    X, y = datasets.make_classification(n_samples=10000, random_state=42)
```

Comment:
Yes! The `ALL_TREES[...]` contains regressors too... Just that I use the same classification data to test the regressors too...